Render individual compilation sections in `--timings` pipeline graph

Kobzol opened the pull request:
### What does this PR try to resolve?
This PR expands the support for `--json-timings` (added in https://github.com/rust-lang/cargo/pull/15780), by rendering the individual compilation sections in the pipeline graph of the `--timings` page.

Before, the linking section was only shown in the table at the bottom of the page, now it should also be clearly visible in the compilation graph, which should help more quickly understand how much time is spent in linking.

<img width="1219" height="358" alt="image" src="https://github.com/user-attachments/assets/71d0200d-4175-43b7-8aab-997008e2af47" />

I also added a legend to the pipeline graph, to explain what do the colors mean.
<img width="338" height="118" alt="image" src="https://github.com/user-attachments/assets/69f9bac6-c33f-44c5-8e45-afa97f314e4c" />

One wart is that the linking time actually ends a bit before the unit ends, so there is some "vacuum" at the end where rustc does cleanup, persists files to disk, deallocates things, etc. That's why I marked the blue section "Frontend/rest" in the legend.

### How to test and review this PR?
Same as for https://github.com/rust-lang/cargo/pull/15780, e.g.:
```bash
export RUSTC=`rustup +nightly which rustc`
target/debug/cargo build -Zsection-timings --timings
```
on some crate, e.g. [ripgrep](https://github.com/BurntSushi/ripgrep).

Kobzol commented:
Could be. We could create an artificial "Other" section, which would be from the end of the last known section to the end of the compilation of the unit (as long as this duration is non-empty, of course). Does that work?

epage commented on src/cargo/core/compiler/timings.js:
```diff
@@ -188,7 +201,9 @@ function render_pipeline_graph() {
-    } else {
+    } else if (name === "link") {
+        return LINK_COLOR;
+    } {
```
`else`?

epage commented:
> Could be. We could create an artificial "Other" section, which would be from the end of the last known section to the end of the compilation of the unit (as long as this duration is non-empty, of course). Does that work?

Works for me!

weihanglo reviewed (commented):
This is nice. Do you have a sense of when the rustc flag will be stabilized?

Kobzol commented:
I mean, I could propose stabilization immediately, but I'm not sure if it's a good idea yet. For example, @bjorn3 is currently doing some changes of how LTO works, and while doing that, we found out that you can have multiple codegen + linking sections during a single rustc invocation. That kind of breaks the assumption that `--timings` makes (even without taking sections into account).

Kobzol commented:
Fixed the missing `else` and added the "other" section. I can imagine improving this page further (e.g. adding a tooltip on hover on a crate that would show the names and durations of the individual sections), but it's kinda bothersome with the raw JavaScript and the canvas :/

epage reviewed (approved):
Thanks!
